### PR TITLE
adding basic http proxy support

### PR DESCRIPTION
**Pull request by solset-sjohnson**
This allows the client to make requests to kaltura api when a http proxy is in place on the host machine.

### CODE REVIEW

**Comment by tan-tan-kanarek** [comment]
Thank you for your contribution.
Could you please move the proxy properties to KalturaConfiguration and later access them from KalturaClientBase using this.config?
I believe that proxy properties should be defined as part of the configuration.

**Comment by solset-sjohnson** [comment]
Good idea, PR is updated.

File: sources/node/KalturaClientBase.js [hunk 1]
```diff
@@ -674,3 +681,12 @@ KalturaConfiguration.prototype.setLogger = function(logger){
 KalturaConfiguration.prototype.getLogger = function(){
 	return this.logger;
 };
+
+/**
+ * @param host - hostname or ip address of proxy
+ * @param port - port of proxy
+ */
+KalturaConfiguration.prototype.setProxy = function(host, port)
```
Comment (yossipapi) [inline sources/node/KalturaClientBase.js]: This method is no longer used. 
Should be removed.

Comment (solset-sjohnson) [inline sources/node/KalturaClientBase.js]: It is optionally used by the application creating the KalturaClient to set proxy values of the host, like below:

`var config = new Kaltura.kc.KalturaConfiguration(partnerID);
config.setProxy('localhost', 3128);
var kalturaClient = new Kaltura.kc.KalturaClient(config);

`

**Pull request merged**